javascript-bank/bank.js

```diff
@@ -0,0 +1,40 @@
+/* exported Bank */
+function Bank() {
+  this.nextAccountNumber = 1;
+  this.accounts = [];
+}
+
+Bank.prototype.openAccount = function (holder, balance) {
+  if (balance > 0 && balance % 1 === 0) {
+    var newAccount = new Account(this.nextAccountNumber, holder);
+    newAccount.deposit(balance);
+    this.accounts.push(newAccount);
+    this.nextAccountNumber++;
+    return newAccount.number;
+  } else {
+    return null;
+  }
+
+};
+
+Bank.prototype.getAccount = function (number) {
+  for (var i = 0; i < this.accounts.length; i++) {
+    if (number === this.accounts[i].number) {
+      return this.accounts[i];
+    }
+  }
+  return null;
+};
+
+Bank.prototype.getTotalAssets = function () {
+  var total = 0;
+  if (this.accounts.length === 0) {
+    return 0;
+  } else {
+    for (var i = 1; i < this.accounts.length + 1; i++) {
+      total += this.getAccount(i).getBalance();
```

Comment:
Since the `getAccount` method can sometimes return `null`, be sure to check whether the account actually exists before attempt to get its balance to avoid errors.

Reply:
hello @TimelikeClosure , I updated the code to exclude any null results from the getAccount method. Let me know if this works!